docs/backend-requests/resources/cookie-size-limits.mdx

```diff
@@ -0,0 +1,12 @@
+---
+title: Cookie size limits
+description: Learn about cookie size limits and how to stay within them with Clerk.
+---
+
+All modern browsers [limit the maximum size of a cookie to 4kb](https://datatracker.ietf.org/doc/html/rfc2109#section-6.3).
+
+The Clerk [session token](/docs/backend-requests/resources/session-tokens) is stored in a Cookie, and is required for Clerk to function. Exceeding the 4kb max size can have adverse effects, including a possible infinite redirect loop for users who exceed this size in Next.js applications.
+
+A session token with the [default session claims](/docs/backend-requests/resources/session-tokens#default-session-claims) will not run into this issue, as this configuration produces a Cookie significantly smaller than 4kb. Only when [customizing your session token](/docs/backend-requests/making/custom-session-token) or creating a [custom JWT template](/docs/backend-requests/making/jwt-templates) do you have to be mindful of this limitation.
+
+It's recommended to move particularly large claims out of the JWT and fetch these using a separate API call from your backend. Notably, the `user.organizations` claim can often trigger the cookie exceeding 4kb.
```

Comment:
```suggestion
It's recommended to move particularly large claims out of the JWT and fetch these using a separate API call from your backend. Notably, the `user.organizations` claim will often cause the cookie to exceed 4kb. Other claims to be cautious about are `user.public_metadata`, `user.unsafe_metadata`, `org.public_metadata`, and `org_membership.public_metadata`. With any of these claims, if enough data is stored as metadata, it will cause the cookie size to exceed the limit. If you are including any of these claims in your JWT, use extra caution around how much is stored in the metadata fields to avoid a cookie size overflow issue.
```